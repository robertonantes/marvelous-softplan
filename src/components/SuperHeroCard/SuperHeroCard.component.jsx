import React from "react";
import { Name, Wrapper } from "./SuperHeroCard.styles";

const SuperHeroCard = ({ name, image }) => {
  return (
    <Wrapper background={image} data-testid="hero-image">
      <Name>{name}</Name>
    </Wrapper>
  );
};

export default SuperHeroCard;
