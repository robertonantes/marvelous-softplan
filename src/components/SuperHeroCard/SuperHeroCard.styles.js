import styled from "styled-components";

export const Wrapper = styled.div`
  height: 300px;
  max-width: 230px;
  background-image: url(${({ background }) => background});
  background-size: cover;
`;

export const Name = styled.h3``;
