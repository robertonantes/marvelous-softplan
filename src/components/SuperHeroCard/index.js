export { default } from "./SuperHeroCard.component";
