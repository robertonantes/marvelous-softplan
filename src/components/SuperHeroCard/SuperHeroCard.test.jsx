import React from "react";
import { render, screen } from "@testing-library/react";
import SuperHeroCard from ".";

const data = {
  image: "image-path",
  name: "Hero name",
};

describe("SuperHeroCard Component", () => {
  it("Renders card info", () => {
    const { asFragment } = render(
      <SuperHeroCard image={data.image} name={data.name} />
    );

    expect(screen.getByTestId("hero-image")).toHaveStyle(
      `background-image: url(image-path)`
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
