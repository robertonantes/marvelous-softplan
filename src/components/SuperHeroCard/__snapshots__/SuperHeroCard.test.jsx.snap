// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuperHeroCard Component Renders card info 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm dzwwRQ"
    data-testid="hero-image"
  >
    <h3
      class="sc-AxirZ ijJWiI"
    >
      Hero name
    </h3>
  </div>
</DocumentFragment>
`;
